### PR TITLE
kill signature_validation switch since type_validation is superior

### PR DESCRIPTION
**Pull request by deathowl**
#173

### CODE REVIEW

File: docs/patching/mock_callable/index.rst [hunk 1]
```diff
@@ -378,20 +378,26 @@ Mocking magic methods (eg: ``__str__``) for an instance can be quite tricky, as
 
 The mock works for the target instance, but does not affect other instances.
 
-Signature Validation
---------------------
 
-mock_callable implements signature validation. When you use it, the mock will raise ``TypeError`` if it is called with a signature that does not match the original method:
+Type Validation
+---------------
+When you use it, the mock will raise ``TypeError`` if it is called with a signature that does not match the original method:
```
Comment (fornellas) [inline docs/patching/mock_callable/index.rst:384]: s/When you use it/By default/

File: docs/patching/mock_callable/index.rst [hunk 2]
```diff
@@ -378,20 +378,26 @@ Mocking magic methods (eg: ``__str__``) for an instance can be quite tricky, as
...
+
+If typing annotation information is available, ``mock_callable()`` validates types of objects passing through the mock. If an invalid type is detected, it will raise ``TypeError``.
+
+This feature is enabled by default. If you need to disable it (potentially due to a bug, please report!), you can do so by ``mock_callable(target, name, type_validation=False)``.
```
Comment (fornellas) [inline docs/patching/mock_callable/index.rst:388]: s/This feature is enabled by default. /

**Comment by deathowl** [comment]
tests seem fine too except minimal decrease in coverage

**Comment by david-caro** [comment]
The line that got uncovered is https://coveralls.io/builds/31862546/source?filename=testslide/strict_mock.py#L592, when we return an exception if the return value is non-awaitable, maybe we can add a test for that?

**Comment by deathowl** [comment]
@david-caro sure. will do it this evening

**Pull request merged**